### ISSUE TITLE
False negative for `Layout/LineLength` with split heredoc

### ISSUE THREAD

**Issue by andyw8** [rubocop/rubocop#12830]
## Expected behavior

Given this code, and max line length set to 120:

```
x = <<~FOO
  a \
  abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef abcdef
FOO
```

RuboCop should complain with `Layout/LineLength: Line is too long. [127/120]`.

## Actual behavior

RuboCop doesn't complain. **But** if I set `ParserEngine: parser_prism`, then it correctly reports the issue.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.63.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-performance 1.18.0
  - rubocop-rspec 2.23.2
```

**Comment by dvandersluis** [comment]
By default, `Layout/LineLength` is configured with `AllowHeredoc: true`, which allows long lines inside heredocs. If you set `AllowHeredoc: false`, an offense will be registered for the second line of your heredoc.

FWIW, I get the same behaviour with prism and with parser (no offense with `AllowHeredoc: true`, offense with `AllowHeredoc: false`).

**Comment by andyw8** [comment]
@dvandersluis thank you for that.

Prism's behaviour may have changed since I first reported this.

**Comment by dvandersluis** [comment]
Going to close this for now but feel free to reopen if you can provide a reproduction case.
